File: Projeto_6/projeto 6 Davi/src/Professor.cpp

#include "../include/Professor.h"
#include <iostream>

Professor::Professor(std::string nome, std::string login, std::string senha)
    : UsuarioAutenticavel(nome, login, senha) {}

bool Professor::autenticar(std::string senhaTentativa) {
    return senha == senhaTentativa;
}

void Professor::gerarRelatorio() const {
    std::cout << "Relatório do Professor: " << nome << std::endl;
}

File: Projeto-7_Kayron/src/professor.py
from .usuarioAutenticavel import UsuarioAutenticavel
from .relatorio import Relatorio
from .tipoUsuario import TipoUsuario


class Professor(UsuarioAutenticavel, Relatorio):
    def __init__(self, nome: str, senha: str, disciplina: str):
        super().__init__(nome, senha)
        self.tipo = TipoUsuario.PROFESSOR
        self.disciplina = disciplina

    def autenticar(self, senha: str) -> bool:
        return self._verificar_senha(senha)

    def gerar_relatorio(self):
        print(f"Relatório do Professor {self.nome}:")
        print(f"Disciplina lecionada: {self.disciplina}")
